### PR TITLE
Email invite

### PR DESCRIPTION
**Pull request by chinmaypatil265**
Added email invitation functionality

### CODE REVIEW

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { Redirect, useParams } from 'react-router';
+import sendEmails from 'modules/admin/sendEmailInvite/SendEmailInviteContainer/sendEmails';
+
+const SendEmailInviteContainer = () => {
+  const [emails, setEmails] = useState('');
+  const [csvEmails, setCsvEmails] = useState('');
+  const [emailsError, setEmailsError] = useState('');
+  const [csvFileError, setCsvFileError] = useState('');
+  const [successMessage, setSuccessMessage] = useState('');
+  const [loading, setLoading] = useState(false);
```
Comment (atharva-borekar) [inline src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:12]: avoid using these many use states, instead use one reducer

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      const text1 = text.replace(/,\s*$/, '');
+      setCsvEmails(text1);
+    } else {
+      setCsvFileError('Invalid file type.Content will be ignored(.csv needed)');
+    }
+  };
+
+  const validateEmails = (checkEmails) => {
+    let flag = false;
+    const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (atharva-borekar) [inline src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:33]: use yup validation for checking format of email or any field

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react';
+import SendEmailInviteComponent from 'modules/admin/sendEmailInvite/SendEmailInviteComponent';
+import { Redirect, useParams } from 'react-router';
+import sendEmails from 'modules/admin/sendEmailInvite/SendEmailInviteContainer/sendEmails';
```
Comment (atharva-borekar) [inline src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:4]: change import order:
1. main libraries
2. dependent libraries
3. components and containers, reducers
4. functions/methods like api call
5. constants

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/sendEmails.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import axios from 'axios';
+import { SERVER_URL } from 'constants/appConstants';
+import local from 'utils/local';
```
Comment (atharva-borekar) [inline src/modules/admin/sendEmailInvite/SendEmailInviteContainer/sendEmails.js:3]: import order: constants should be imported at last

File: src/modules/admin/sendEmailInvite/SendEmailInviteComponent/index.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                  type='file'
+                  name='Selectfile'
+                  id='selectFile'
+                  onChange={handleUploadedInvitationEmails}
+                  invalid={csvFileError !== ''}
+                />
+                <FormFeedback>{csvFileError}</FormFeedback>
+              </FormGroup>
+              <FormGroup className='mt-6'>
+                <Button color='success' className='mr-3 mt-3' onClick={handleSendInvitation}>
```
Comment (Rutu2k) [inline src/modules/admin/sendEmailInvite/SendEmailInviteComponent/index.js:79]: we have reactstrap classes for colors
e.g: className=' bg-success'  and className='text-success' try using them

File: src/modules/admin/sendEmailInvite/SendEmailInviteComponent/index.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                  Send Invitation
+                </Button>
+                <Button type='reset' color='warning' className='ml-3 mr-3 mt-3' onClick={handleCsvRemove}>Clear CSV File</Button>
+                <Button color='danger' className='ml-3 mt-3' onClick={handleCancel}>Cancel</Button>
```
Comment (Rutu2k) [inline src/modules/admin/sendEmailInvite/SendEmailInviteComponent/index.js:83]: Have you checked the case ..submit clicked when fields are empty.if not then validate it.

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js (hunk 2 above)
Comment (Rutu2k) [reply src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:33]: yes implement yup validation

**Review by Rutu2k — COMMENTED** [review]
Let me know if have any doubt

File: src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      let csvEmailText = await e.target.files[0].text();
+      csvEmailText = csvEmailText.replace(/(\n)/gm, ',');
+      const finalCsvEmailText = csvEmailText.replace(/,\s*$/, '');
+      dispatch({ type: 'VALID_CSV_FILETYPE', payload: finalCsvEmailText });
+    } else {
+      dispatch({ type: 'INVALID_CSV_FILETYPE' });
+    }
+  };
+
+  const validateEmails = (checkEmails) => {
```
Comment (Suhas010) [inline src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:27]: can be moved to utilities.

Comment (Suhas010) [reply src/modules/admin/sendEmailInvite/SendEmailInviteContainer/index.js:27]: @chinmaypatil265 As discussed offline, ignore this comment for now.

**Review by Suhas010 — APPROVED** [review]

**Pull request merged**